stratum/hal/lib/tdi/tdi_fixed_function_manager.cc

```diff
@@ -74,70 +99,71 @@ ::util::Status TdiFixedFunctionManager::WriteSadbEntry(
 }
 
 ::util::Status TdiFixedFunctionManager::FetchSpi(
-  std::shared_ptr<TdiSdeInterface::SessionInterface> session,
-  std::string table_name, uint32 *spi) {
-
+    std::shared_ptr<TdiSdeInterface::SessionInterface> session,
+    std::string table_name, uint32* spi) {
   absl::ReaderMutexLock l(&lock_);
 
   uint64 spi_val;
   ASSIGN_OR_RETURN(uint32 table_id, tdi_sde_interface_->GetTableId(table_name));
-  ASSIGN_OR_RETURN(auto table_data, tdi_sde_interface_->CreateTableData(table_id, 0));
+  ASSIGN_OR_RETURN(auto table_data,
+                   tdi_sde_interface_->CreateTableData(table_id, 0));
   RETURN_IF_ERROR(tdi_sde_interface_->GetDefaultTableEntry(
-            device_, session, table_id, table_data.get()));
+      device_, session, table_id, table_data.get()));
   RETURN_IF_ERROR(table_data->GetParam(kIpsecFetchSpi, &spi_val));
-  *spi = (uint32) spi_val;
+  *spi = (uint32)spi_val;
   return ::util::OkStatus();
 }
 
 ::util::Status TdiFixedFunctionManager::BuildSadbTableKey(
-  TdiSdeInterface::TableKeyInterface* table_key,
-  IPsecSADBConfig &sadb_config) {
-
-    CHECK_RETURN_IF_FALSE(table_key);
-
-    RETURN_IF_ERROR(table_key->SetExact(
-      kIpsecSadbOffloadId, sadb_config.offload_id()));
-    RETURN_IF_ERROR(table_key->SetExact(
-      kIpsecSadbDir, ((uint8)sadb_config.direction())));
-    return ::util::OkStatus();
+    TdiSdeInterface::TableKeyInterface* table_key,
+    IPsecSADBConfig& sadb_config) {
+  CHECK_RETURN_IF_FALSE(table_key);
+
+  RETURN_IF_ERROR(
+      table_key->SetExact(kIpsecSadbOffloadId, sadb_config.offload_id()));
+  RETURN_IF_ERROR(
+      table_key->SetExact(kIpsecSadbDir, ((uint8)sadb_config.direction())));
+  return ::util::OkStatus();
 }
 
 ::util::Status TdiFixedFunctionManager::BuildSadbTableData(
-  TdiSdeInterface::TableDataInterface* table_data,
-  IPsecSADBConfig &sadb_config) {
-
-    CHECK_RETURN_IF_FALSE(table_data);
-
-    RETURN_IF_ERROR(table_data->SetParam(kIpsecSadbReqId, sadb_config.req_id()));
-    RETURN_IF_ERROR(table_data->SetParam(kIpsecSadbSpi, sadb_config.spi()));
+    TdiSdeInterface::TableDataInterface* table_data,
+    IPsecSADBConfig& sadb_config) {
+  // Wrapper object ensures that sadb_config is scrubbed on exit.
```

Comment:
Can we say IPsec key is zeroed instead of scrubbed?

Reply:
It doesn't matter what we write to the location. I didn't want the abstract interface to be tied to the value used.
I will substitute the term "sanitized", which is used to describe this operation in other contexts.

Reply:
Okay makes sense, please change the title as well.